### PR TITLE
Feature/about translation ts

### PR DESCRIPTION
**Pull request by noctillion**
Description:
This PR adds support for translating the about.html content in the Overview tab. The changes introduced include:

Changes:

Basic templates eng_about.html and fr_about.html
Fetching both English and French versions of x_about.html content using async thunk.
Dynamically rendering the x_about.html content in the Overview tab based on the active language.

### CODE REVIEW

File: src/js/constants/contentConstants.ts [hunk 1]
```diff
@@ -1 +1 @@
-export const aboutUrl = '/public/about.html';
+export const partialAboutUrl = '/public/';
```
Comment (v-rocheleau) [inline src/js/constants/contentConstants.ts:1]: Replace `'/public/'` with `'/public'`. All of our url constants have no trailling slash.

File: src/js/features/content/content.store.ts [hunk 2]
```diff
@@ -1,25 +1,32 @@
...
-      .catch(printAPIError(rejectWithValue))
-);
+export const makeGetAboutRequest = createAsyncThunk('content/getAboutHTML', async () => {
+  const en_aboutHTML = await axios
+    .get(`${partialAboutUrl}en_about.html`)
+    .then((res) => res.data)
+    .catch(printAPIError);
+
+  const fr_aboutHTML = await axios
+    .get(`${partialAboutUrl}fr_about.html`)
```
Comment (v-rocheleau) [inline src/js/features/content/content.store.ts:13]: Add `/` before file name once partialAboutUrl has no trailling slash

File: src/js/features/content/content.store.ts [hunk 3]
```diff
@@ -1,25 +1,32 @@
...
+
+  return { en_aboutHTML, fr_aboutHTML };
+});
 
 export type ContentState = {
   isFetchingAbout: boolean;
-  aboutHTML: string;
+  en_aboutHTML: string;
+  fr_aboutHTML: string;
 };
```
Comment (v-rocheleau) [inline src/js/features/content/content.store.ts:24]: I would change the ContentState shape for:
```typescript
export type ContentState = {
  isFetchingAbout: boolean;
  about: {[key: string]: string}
};

const initialState: ContentState = {
  isFetchingAbout: true,
  about: {
    en: "",
    fr: ""
  }
};
```
This way, we can get the about content by key and avoid the if blocks in PublicOverview's useEffect:
```typescript
  const { isFetchingAbout, about } = useAppSelector((state) => state.content);

  const [aboutContent, setAboutContent] = useState('');

  useEffect(() => {
    const activeLanguage = i18n.language;
    const activeAbout = about[activeLanguage];
    setAboutContent(activeAbout);
  }, [i18n.language, about]);
```
Like this we won't have to change the PublicOverview component if we add new languages

**Review by v-rocheleau — CHANGES_REQUESTED** [review]

**Comment by noctillion** [comment]
Thanks for the review and excellent suggestions!

File: src/js/features/content/content.store.ts [hunk 4]
```diff
@@ -30,8 +38,9 @@ const content = createSlice({
     builder.addCase(makeGetAboutRequest.pending, (state) => {
       state.isFetchingAbout = true;
     });
-    builder.addCase(makeGetAboutRequest.fulfilled, (state, { payload }: PayloadAction<string>) => {
-      state.aboutHTML = payload;
+    builder.addCase(makeGetAboutRequest.fulfilled, (state, { payload }: PayloadAction<{ en: string; fr: string }>) => {
+      state.about.en = payload.en;
+      state.about.fr = payload.fr;
```
Comment (v-rocheleau) [inline src/js/features/content/content.store.ts:43]: State update can be simplified with:
```typescript
state.about = {...payload};
```

**Review by v-rocheleau — CHANGES_REQUESTED** [review]

File: src/js/features/content/content.store.ts [hunk 5]
```diff
@@ -39,8 +39,7 @@ const content = createSlice({
       state.isFetchingAbout = true;
     });
     builder.addCase(makeGetAboutRequest.fulfilled, (state, { payload }: PayloadAction<{ en: string; fr: string }>) => {
-      state.about.en = payload.en;
-      state.about.fr = payload.fr;
+      state.about = { ...payload}
```
Comment (v-rocheleau) [inline src/js/features/content/content.store.ts:42]: Missing semicolon

**Review by v-rocheleau — APPROVED** [review]

**Pull request merged**